pkg/common/providers/ocmprovider/cluster.go

```diff
@@ -392,7 +392,7 @@ func (o *OCMProvider) ListClusters(query string) ([]*spi.Cluster, error) {
 	emptyPage := false
 	page := 1
 
-	for len(clusters) < totalItems || emptyPage {
```

Comment:
I've just modified this code snippet to stop looping if there are no results for the corresponding search string or when there are no more results to bring in. As of now, the variable emptyPage seems to serve this operation. I can revert this change if this impacts overall operation.